### PR TITLE
Run linters in GitHub action

### PR DESCRIPTION
**Pull request by Lee-W**
* github-action: run linter, test and coverage
* makefile
    * add lint, format, test, coverage targets
* poetry: update dev packages
* style(config): init mypy config
* style: reformat

### CODE REVIEW

File: .github/workflows/python.yml [hunk 1]
```diff
@@ -21,12 +19,21 @@ jobs:
       with:
         python-version: 3.7
 
-    - name: Run test
+    - name: Install dependencies
```
Comment (david30907d) [inline .github/workflows/python.yml:22]: thanks haha

File: Makefile [hunk 2]
```diff
@@ -0,0 +1,17 @@
+VENV_PREFIX=poetry run
```
Comment (david30907d) [inline Makefile:1]: nice~

File: setup.cfg [hunk 3]
```diff
@@ -0,0 +1,7 @@
+[mypy]
```
Comment (david30907d) [inline setup.cfg:1]: cool~ didn't know that lol

**Review by david30907d — COMMENTED** [review]
LGTM except the bigquery import 

i guess is that i'm using `isort -rc -y -ac -ls` and you just use pure `isort`, not sure

**Comment by Lee-W** [comment]
> LGTM except the bigquery import
> 
> i guess the root cause is that i'm using `isort -rc -y -ac -ls` and you just use pure `isort`, not sure

Not so sure. But my best guess is that I use `profile=black` in `pyproject.toml`. Also, I'd like to know which isort version are you running? It seems that I don't have the `-y` argument.
I'm running isort=5.6.4

```

                 _                 _
                (_) ___  ___  _ __| |_
                | |/ _/ / _ \/ '__  _/
                | |\__ \/\_\/| |  | |_
                |_|\___/\___/\_/   \_/

      isort your imports, so you don't have to.

                    VERSION 5.6.4
```

**Comment by david30907d** [comment]
my `isort is 4.3.21`

<https://github.com/pycontw/PyCon-ETL/blob/master/pyproject.toml#L17>

and i also apply `black` in [package.json](https://github.com/pycontw/PyCon-ETL/blob/master/package.json#L23) so maybe it's not the root cause?

anyway, i'm okay with using the latest `isort`, let me remove `-y` in `package.json`

**Review by david30907d — APPROVED** [review]
Thanks for this PR~~

**Pull request merged**